scripts/dist.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+
+# This script builds a dist tarball of the GD2 source
+# This should only be called from the root of the GD2 repo
+
+VENDOR=${VENDOR:-no}
+OUTDIR=${1:-.}
+
+VERSION=$($(dirname $0)/pkg-version --full)
+
+TARNAME=glusterd2-$VERSION
+case $VENDOR in
+  yes|y|Y)
+    TARNAME+="-vendor"
+    ;;
+
+  *)
+    EXCLUDEVENDOR="--exclude vendor"
+    ;;
+esac
+
+TARBALL=$TARNAME.tar
+ARCHIVE=$OUTDIR/$TARBALL.xz
+SIGNFILE=$ARCHIVE.asc
+
+# Cleanup old archives
+if [[ -f $ARCHIVE ]]; then
+  rm $ARCHIVE
+fi
+if [[ -f $SIGNFILE ]]; then
+  rm $SIGNFILE
+fi
+
+# Create the VERSION file first
+$(dirname $0)/gen-version.sh
+
+# Use a temp dir to build tar file
+TMPDIR=$(mktemp -d)
+
+echo "Creating dist tarball $ARCHIVE"
+# Exclude .git, releases, builds and older tarballs
+tar --exclude .git --exclude releases --exclude build --exclude 'glusterd2-*.tar.xz*' $EXCLUDEVENDOR --transform "s/^\./$TARNAME/" -cf $TMPDIR/$TARBALL . || exit 1
```

Comment:
This can become problematic when these list of dirs/files to be excluded grows or is different on my machine. Would you prefer to use `git-archive` instead which is cleaner ?

Reply:
The reason for creating the dist command was to also package vendored packages. The vendored packages aren't in git, so git-archive wouldn't contain them. So I started from with this to archive everything, and went back excluding things as required.

It should be possible create a git-archive tar and then add the vendored package to it. I'll see what I can do.

Reply:
In the new commits, did you intentionally leave out including the `VERSION` file from tarball ?

Reply:
That was unintended.